spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/RetrievalAugmentationAdvisor.java

```diff
@@ -119,30 +127,45 @@ private AdvisedRequest before(AdvisedRequest request) {
 		Map<String, Object> context = new HashMap<>(request.adviseContext());
 
 		// 0. Create a query from the user text and parameters.
-		Query query = new Query(new PromptTemplate(request.userText(), request.userParams()).render());
+		Query originalQuery = new Query(new PromptTemplate(request.userText(), request.userParams()).render());
+
+		// 1. Transform original user query based on a chain of query transformers.
+		Query transformedQuery = originalQuery;
+		for (var queryTransformer : queryTransformers) {
+			transformedQuery = queryTransformer.apply(transformedQuery);
+		}
 
-		// 1. Retrieve similar documents for the original query.
-		List<Document> documents = this.documentRetriever.retrieve(query);
+		// 2. Retrieve similar documents for the original query.
+		List<Document> documents = this.documentRetriever.retrieve(transformedQuery);
 		context.put(DOCUMENT_CONTEXT, documents);
 
-		// 2. Augment user query with the contextual data.
-		Query augmentedQuery = this.queryAugmentor.augment(query, documents);
+		// 3. Augment user query with the document contextual data.
+		Query augmentedQuery = this.queryAugmentor.augment(transformedQuery, documents);
 
 		return AdvisedRequest.from(request).withUserText(augmentedQuery.text()).withAdviseContext(context).build();
```

Comment:
wondering about the multimodal/media aspects of this as it focused on text and any media for the request is lost.

Reply:
The AdvisedRequest has a separate Media property for the media elements, so they are currently propagated correctly.